presto-main/src/main/java/com/facebook/presto/operator/project/ConstantPageProjection.java

```diff
@@ -62,27 +61,31 @@ public InputChannels getInputChannels()
     }
 
     @Override
-    public PageProjectionOutput project(ConnectorSession session, DriverYieldSignal yieldSignal, Page page, SelectedPositions selectedPositions)
+    public Work<Block> project(ConnectorSession session, DriverYieldSignal yieldSignal, Page page, SelectedPositions selectedPositions)
     {
-        return new ConstantPageProjectionOutput(value, selectedPositions.size());
+        return new ConstantPageProjectionWork(value, selectedPositions.size());
     }
 
-    private class ConstantPageProjectionOutput
-            implements PageProjectionOutput
+    private class ConstantPageProjectionWork
+            implements Work<Block>
     {
-        private final Block value;
-        private final int size;
+        private final RunLengthEncodedBlock result;
 
-        public ConstantPageProjectionOutput(Block value, int size)
+        public ConstantPageProjectionWork(Block value, int size)
+        {
+            this.result = new RunLengthEncodedBlock(value, size);
```

Comment:
I believe it makes more sense to have a vanilla ConstantWork<T>. You can do `new ConstantWork<>(new RunLengthEncodedBlock(value, selectedPositions.size()))` instead.

```
class CompletedWork<T>
        implements Work<T> {
    private final T result;

    public ConstantWork(T value) {
        this.result = requireNonNull(value);
    }

    public boolean process() {
        return true;
    }

    public Block getResult() {
        return result;
    }
}
```

Considering this class will also be useful in DictionaryAwarePageProjection line 200, I would want to put this in a more accessible location. I can think of two choices.
* I would make CompletedWork an anonymous class in a static method in Work interface. The method can be called `completedWork`. (Use capture instead of constructor if you make it an anonymous class.) Remember to statically import it when you use it.
* But you can also make it a stand-alone class under com.facebook.presto.operator if that's what you prefer.